src/main/java/org/apache/accumulo/testing/cli/ClientOpts.java

```diff
@@ -0,0 +1,213 @@
+
+  public AuthenticationToken getToken() {
+    return ClientProperty.getAuthenticationToken(getClientProperties());
+  }
+
+  @Parameter(names = {"-z", "--keepers"},
+      description = "Comma separated list of zookeeper hosts (host:port,host:port)")
+  protected String zookeepers = null;
+
+  @Parameter(names = {"-i", "--instance"}, description = "The name of the accumulo instance")
+  protected String instance = null;
+
+  @Parameter(names = {"-auths", "--auths"}, converter = AuthConverter.class,
+      description = "the authorizations to use when reading or writing")
+  public Authorizations auths = Authorizations.EMPTY;
+
+  @Parameter(names = "--debug", description = "turn on TRACE-level log messages")
+  public boolean debug = false;
+
+  @Parameter(names = "--ssl", description = "Connect to accumulo over SSL")
+  private boolean sslEnabled = false;
+
+  @Parameter(names = "--sasl", description = "Connecto to Accumulo using SASL (supports Kerberos)")
+  private boolean saslEnabled = false;
+
+  @Parameter(names = "--config-file", description = "Read the given client config file. "
+      + "If omitted, the classpath will be searched for file named accumulo-client.properties")
+  private String clientConfigFile = null;
+
+  public void startDebugLogging() {
+    if (debug)
+      Logger.getLogger("org.apache.accumulo.testing").setLevel(Level.TRACE);
+  }
+
+  @Parameter(names = "--trace", description = "turn on distributed tracing")
+  public boolean trace = false;
+
+  @Parameter(names = "--keytab", description = "Kerberos keytab on the local filesystem")
+  private String keytabPath = null;
+
+  @Override
+  public void parseArgs(String programName, String[] args, Object... others) {
+    super.parseArgs(programName, args, others);
+    startDebugLogging();
+  }
+
+  private Properties cachedProps = null;
+
+  public String getPrincipal() {
+    return ClientProperty.AUTH_PRINCIPAL.getValue(getClientProperties());
+  }
+
+  public void setPrincipal(String principal) {
+    this.principal = principal;
+  }
+
+  public void setClientProperties(Properties clientProps) {
+    this.cachedProps = clientProps;
+  }
+
+  /**
+   * @return {@link AccumuloClient} that must be closed by user
+   */
+  public AccumuloClient createClient() {
+    return Accumulo.newClient().from(getClientProperties()).build();
+  }
+
+  public String getClientConfigFile() {
+    if (clientConfigFile == null) {
+      URL clientPropsUrl = ClientOpts.class.getClassLoader().getResource(
+          "accumulo-client.properties");
+      if (clientPropsUrl != null) {
+        clientConfigFile = clientPropsUrl.getFile();
+      }
+    }
+    return clientConfigFile;
+  }
+
+  public Properties getClientProperties() {
+    if (cachedProps == null) {
+      cachedProps = new Properties();
+      if (getClientConfigFile() != null) {
+        cachedProps = toProperties(Paths.get(getClientConfigFile()));
+      }
+      if (saslEnabled) {
```

Comment:
this could be a follow on PR, but looking at this made me think of it.  I think it would be nice to remove any command line options that can be set in the client props.  Also could support a command line option to set a client prop.  I recently did this in #51 for continuous ingest code.  Being able to set client props on the command line is great for docker.

Reply:
Nice idea. I look at doing this in another PR.